src/configuration.cpp

```diff
@@ -0,0 +1,702 @@
+                   .group("Link & Install")
+                   .set_env_var_name()
+                   .description("Safety checks policy ('enabled', 'warn', or 'disabled')")
+                   .long_description(unindent(R"(
+                    Enforce available safety guarantees during package installation. The
+                    value must be one of 'enabled', 'warn', or 'disabled'.)")));
+
+        insert(Configurable("extra_safety_checks", &ctx.extra_safety_checks)
+                   .group("Link & Install")
+                   .set_env_var_name()
+                   .description("Run extra verifications on packages")
+                   .long_description(unindent(R"(
+                    Spend extra time validating package contents. Currently, runs sha256
+                    verification on every file within each package during installation.)")));
+
+        // Output, Prompt and Flow
+        insert(Configurable("auto_activate_base", &ctx.auto_activate_base)
+                   .group("Output, Prompt and Flow Control")
+                   .set_env_var_name()
+                   .description("Automatically activate the base env")
+                   .long_description(unindent(R"(
+                    Automatically activate the base environment during shell
+                    initialization.)")));
+    }
+
+    void Configuration::reset_configurables()
+    {
+        m_config.clear();
+        m_config_order.clear();
+        set_configurables();
+    }
+
+    std::vector<std::pair<std::string, std::vector<ConfigurableInterface*>>>
+    Configuration::get_grouped_config()
+    {
+        std::map<std::string, std::vector<ConfigurableInterface*>> map;
+        std::vector<std::pair<std::string, std::vector<ConfigurableInterface*>>> res;
+        std::vector<std::string> group_order;
+
+        for (auto& name : m_config_order)
+        {
+            auto& c = m_config.at(name);
+
+            if (map.count(c.group()) == 0)
+            {
+                group_order.push_back(c.group());
+            }
+            map[c.group()].push_back(&c);
+        }
+
+        for (auto& g : group_order)
+        {
+            res.push_back({ g, map.at(g) });
+        }
+
+        return res;
+    }
+
+    void Configuration::set_possible_sources()
+    {
+        auto& ctx = Context::instance();
+
+        fs::path condarc_env_var = std::getenv("CONDARC") ? std::getenv("CONDARC") : "";
+        fs::path mambarc_env_var = std::getenv("MAMBARC") ? std::getenv("MAMBARC") : "";
+
+        if (on_mac || on_linux)
+        {
+            m_sources = { "/etc/conda/.condarc",       "/etc/conda/condarc",
+                          "/etc/conda/condarc.d/",     "/etc/conda/.mambarc",
+                          "/var/lib/conda/.condarc",   "/var/lib/conda/condarc",
+                          "/var/lib/conda/condarc.d/", "/var/lib/conda/.mambarc" };
+        }
+        else
+        {
+            m_sources = { "C:/ProgramData/conda/.condarc",
```

Comment:
Does this work? Usually double backslash is more "correct" on windows.

Reply:
I will change that and maybe add some tests for Windows

Reply:
fixed, not tested yet